### PR TITLE
Implemented MSE 

### PR DESCRIPTION
**Pull request by YashitaBansal**
INPUT:
`std::vector<double> actual = {1.0, 2.0, 3.0}
  std::vector<double> pred = {0.5, 1.5, 2.5}`

OUPTUT:
Mean Squared Error: 0.25

### CODE REVIEW

File: src/slowmokit/methods/metrics/mean_squared_error.cpp [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ *
+ * Implementation of the MSE main program
+ */
+#include "mean_squared_error.hpp"
+template<class T>
+double meanSquaredError(std::vector<T> &actual, std::vector<T> &pred)
+{
+  if (actual.size() != pred.size())
+  {
+    std::cerr << "Warning: actual and pred must have the same length."
```
Comment (uttammittal02) [inline src/slowmokit/methods/metrics/mean_squared_error.cpp:12]: Instead of a warning, throw an error here

**Review by uttammittal02 — CHANGES_REQUESTED** [review]
Also, add docs for all the PRs merged till now and for this one.

**Review by ken1000minus7 — APPROVED** [review]
Lgtm, just make the changes asked by uttam

**Review by harshjohar — APPROVED** [review]
Lgtm

**Comment by harshjohar** [comment]
Fix the clang-format

File: docs/methods/metrics/mean_squared_error.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# Mean Squared Error
+
+The Mean Squared Error (MSE) or Mean Squared Deviation (MSD) of an estimator measures the average of error squares i.e. the average squared difference between the estimated values and true value.
+## Parameters
+
+| Name          | Definition                                                                                  | Type            |
+| ------------- | ------------------------------------------------------------------------------------------- | ----------------|
+| Pred          | Takes a vector of predicted values                                                          | `int,double`    |
+| Actual        | Determines the step size at each iteration while trying to minimize the loss function       | `int,double`    |
```
Comment (uttammittal02) [inline docs/methods/metrics/mean_squared_error.md]: It was intended that you'll change the definition according to your usecase

File: src/slowmokit/methods/metrics/mean_squared_error.hpp [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+#ifndef SLOWMOKIT_MEAN_SQUARED_ERROR_HPP
+#define SLOWMOKIT_MEAN_SQUARED_ERROR_HPP
+#include "../../core.hpp"
+
+/**
+ * Takes predicted and actual values
+ * @param pred -> predicted values
+ * @param actual -> true values
+ * @returns mean squared error
+ * @throws warning and returns -1 in case length of true and predicted vectors
```
Comment (uttammittal02) [inline src/slowmokit/methods/metrics/mean_squared_error.hpp:16]: throws error

**Review by uttammittal02 — CHANGES_REQUESTED** [review]
I'll recommend you to add docs for the previous PRs also in this one itself

File: docs/methods/metrics/mean_squared_error.md (hunk 2 above)
Comment (YashitaBansal) [reply docs/methods/metrics/mean_squared_error.md]: Done sir

File: src/slowmokit/methods/metrics/mean_squared_error.hpp (hunk 3 above)
Comment (YashitaBansal) [reply src/slowmokit/methods/metrics/mean_squared_error.hpp]: done sir

**Review by uttammittal02 — CHANGES_REQUESTED** [review]
Add docs for Precision and recall also

**Review by uttammittal02 — APPROVED** [review]
Lgtm

**Pull request merged**